src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js

```diff
@@ -14,22 +16,49 @@ class CreateBaselineModal extends Component {
         this.submitBaselineName = this.submitBaselineName.bind(this);
 
         this.state = {
-            baselineName: ''
+            baselineName: '',
+            fromScratchChecked: true,
+            copyBaselineChecked: false,
+            copySystemChecked: false
         };
 
         this.updateBaselineName = value => {
             this.setState({ baselineName: value });
         };
+
+        this.handleChecked = (_, event) => {
+            const { value } = event.currentTarget;
+
+            if (value === 'fromScratchChecked') {
+                this.props.clearSelectedBaselines();
+                this.setState({ fromScratchChecked: true, copyBaselineChecked: false, copySystemChecked: false });
+            } else if (value === 'copyBaselineChecked') {
+                this.setState({ fromScratchChecked: false, copyBaselineChecked: true, copySystemChecked: false });
+            } else {
+                this.props.clearSelectedBaselines();
+                this.setState({ fromScratchChecked: false, copyBaselineChecked: false, copySystemChecked: true });
+            }
+        };
     }
 
     async submitBaselineName() {
-        const { baselineName } = this.state;
-        const { createBaseline, toggleCreateBaselineModal } = this.props;
+        const { baselineName, fromScratchChecked } = this.state;
+        const { createBaseline, toggleCreateBaselineModal, selectedBaselineIds, entities } = this.props;
+        let newBaselineObject;
+
         /*eslint-disable camelcase*/
-        let newBaselineObject = { display_name: baselineName, baseline_facts: []};
```

Comment:
If you kept this but dropped the last param then the if-else would be a bit cleaner I think:

```
        if (fromScratchChecked) {
            newBaselineObject['baseline_facts'] =  [];
            await createBaseline(newBaselineObject);
        } else if (selectedBaselineIds.length === 1) {
            await createBaseline(newBaselineObject, selectedBaselineIds[0]);
        } else if (entities.selectedSystemIds.length === 1) {
            newBaselineObject['inventory_uuid'] = entities.selectedSystemIds[0];
            await createBaseline(newBaselineObject);
        }
```